File: rust-examples/examples/ch2_05_standardize_features.rs
use burn::backend::NdArray;
use burn::tensor::Tensor;

type B = NdArray;

fn main() {
    let dev = Default::default();

    // x: [n_samples, n_features]
    let x = Tensor::<B, 2>::from_floats([[1.0, 2.0], [3.0, 4.0], [5.0, 6.0]], &dev);

    let mean = x.clone().mean_dim(0); // [1, features]
    println!("mean = {}", mean.to_data());

    let std = x.clone().var(0).sqrt(); // sqrt of the variance
    println!("std  = {}", std.to_data());

    let z = (x - mean) / std; // broadcast
    println!("z    =\n{}", z.to_data());
}

File: python-examples/ch2_05_standardize_features.py
import torch

x = torch.tensor([[1., 2.], [3., 4.], [5., 6.]])

mean = x.mean(dim=0, keepdim=True)        # [[3., 4.]]
std  = x.var(dim=0, keepdim=True).sqrt()  # [[2., 2.]]   (unbiased, /(n-1))
z    = (x - mean) / std                   # [[-1,-1], [0,0], [1,1]]
